datura/tools/twitter/twitter_advanced_search_tool.py

```diff
@@ -0,0 +1,120 @@
+import json
+import time
+from typing import Type
+import bittensor as bt
+from neurons.validators.apify.twitter_scraper_actor import TwitterScraperActor
+from pydantic import BaseModel, Field
+from starlette.types import Send
+from datura.tools.base import BaseTool
+from datura.services.twitter_prompt_analyzer import TwitterPromptAnalyzer
+from datura.dataset.date_filters import get_specified_date_filter, DateFilterType
+
+
+class TwitterAdvancedSearchToolSchema(BaseModel):
+    query: str = Field(
+        ...,
+        description="The search query for tweets.",
+    )
+
+
+class TwitterAdvancedSearchTool(BaseTool):
+    """Tool that gets tweets from Twitter."""
+
+    name = "Twitter Advanced Search"
+
+    slug = "get_tweets"
+
+    description = "Get tweets for a given query."
+
+    args_schema: Type[TwitterAdvancedSearchToolSchema] = TwitterAdvancedSearchToolSchema
+
+    tool_id = "e831f03f-3282-4d5c-ae01-d7274515194d"
+
+    def _run():
+        pass
+
+    async def _arun(
+        self,
+        query: str,  # run_manager: Optional[CallbackManagerForToolRun] = None
+    ) -> str:
+        """Tweet message and return."""
+        start_time = time.time()
+        date_filter = (
+            self.tool_manager.date_filter
+            if self.tool_manager
+            else get_specified_date_filter(DateFilterType.PAST_WEEK)
+        )
+        start_date = date_filter.start_date.date()
+        end_date = date_filter.end_date.date()
+
+        client = TwitterScraperActor()
+
+        tweets = await client.get_tweets_advanced(
+            urls=[
+
+            ],
+            start=start_date,
+            end=end_date,
+            searchTerms=["Crypto"],
```

Comment:
Hardcoded search terms.

Reply:
Is not finished yet, just run in development mode